go/vt/vttablet/tabletserver/state_manager.go

```diff
@@ -715,6 +716,10 @@ func (sm *stateManager) Broadcast() {
 	defer sm.mu.Unlock()
 
 	lag, err := sm.refreshReplHealthLocked()
+	if sm.demotePrimaryStalled {
+		// If we are stalled while demoting primary, we should send an error for it.
+		err = vterrors.Errorf(vtrpcpb.Code_FAILED_PRECONDITION, "Failed to complete primary demotion")
```

Comment:
hm could we get a more "first class" signal added to the health event, so we don't have to parse out this error message? we don't want to throw away a tablet just because it has a non-nil `err`, we probably want to be able to do something like `if healthEvent.DemotePrimaryStalled { throwAwayTablet() }`.

Reply:
I'd rather not do a proto message change for this. A compromise might be for us to define that error message as a constant, and you can check for that constant before taking action.

Reply:
that works for me! as long as we treat the error message as part of the API surface area, I have no concerns.

Reply:
Fixed